examples/sigstore-go-signing/main.go

```diff
@@ -83,79 +86,87 @@ func main() {
 
 	opts := sign.BundleOptions{}
 
-	// Get trusted_root.json
-	fetcher := fetcher.DefaultFetcher{}
-	fetcher.SetHTTPUserAgent(util.ConstructUserAgent())
+	var signingConfig *root.SigningConfig
 
-	tufOptions := &tuf.Options{
-		Root:              tuf.StagingRoot(),
-		RepositoryBaseURL: tuf.StagingMirror,
-		Fetcher:           &fetcher,
-	}
-	tufClient, err := tuf.New(tufOptions)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	trustedRoot, err := root.GetTrustedRoot(tufClient)
-	if err != nil {
-		log.Fatal(err)
-	}
+	if *signingconfigJSONpath != "" {
+		signingConfig, err = root.NewSigningConfigFromPath(*signingconfigJSONpath)
+		// NOTE: trusted root is not set here (could be taken with another flag)
```

Comment:
In this case, we would no longer pull in a trusted root, which IIRC is now required for signing to verify the results. Could we use this same pattern of providing either a flag or pulling from TUF, something like:

```
if signingconfigJSONPath != "" {
   // read from file
} else {
  // read from TUF
}

if trustedRootJSONPath != "" {
  // read from file
} else {
 // read from TUF
}


```

Reply:
> In this case, we would no longer pull in a trusted root, which IIRC is now required for signing to verify the results.

The signing works, it just does not verify in this case.

> Could we use this same pattern of providing either a flag or pulling from TUF,

I think this suggestion does not work: if user gives us a random signingconfig we should not expect staging trustedroot to verify that bundle.  Trust root and signing config should always come from same source.


I can add another flag so user could provide a trust root as well.